CMakeLists.txt

```diff
@@ -323,25 +323,48 @@ set(CPACK_COMPONENT_INCLUDE_TOPLEVEL_DIRECTORY TRUE)
 # Compress package in parallel.
 set(CPACK_THREADS 0 CACHE STRING "")
 
-
-if(CMAKE_OSX_ARCHITECTURES)
-    # For universal binaries don't include architecture in the package name.
-    set(CPACK_SYSTEM_NAME "${CMAKE_SYSTEM_NAME}")
+# set processor_name
+string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} processor_name)
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```

Comment:
This is used multiple times, should we introduce a variable to tell if we are running on macOS or not, to have simpler checks in the code: if(NEW_MACOS_VAR_NAME) ...

Reply:
I feel the extra variable would currently more overhead than it's worth as we're only using it twice right now, and when introducing the variable you now need to look up the variable to understand this piece of code.

But I don't have strong opinions about it. I'll change it if you think it's valuable and worth the effort.